test/it/uk/gov/hmrc/exports/migrations/changelogs/notification/MakeParsedDetailsOptionalSpec.scala

```diff
@@ -0,0 +1,125 @@
+package uk.gov.hmrc.exports.migrations.changelogs.notification
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.mongodb.{MongoClient, MongoClientURI}
+import com.mongodb.client.{MongoCollection, MongoDatabase}
+import org.bson.Document
+import org.scalatest.{BeforeAndAfterEach, MustMatchers, WordSpec}
+import org.scalatestplus.play.guice.GuiceOneServerPerSuite
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+import stubs.TestMongoDB
+import stubs.TestMongoDB.mongoConfiguration
+import MakeParsedDetailsOptionalSpec._
+
+class MakeParsedDetailsOptionalSpec extends WordSpec with MustMatchers with GuiceOneServerPerSuite with BeforeAndAfterEach {
+
+  override def fakeApplication(): Application =
+    new GuiceApplicationBuilder()
+      .disable[com.kenshoo.play.metrics.PlayModule]
+      .configure(mongoConfiguration)
+      .build()
+
+  private val MongoURI = mongoConfiguration.get[String]("mongodb.uri")
+  private val DatabaseName = TestMongoDB.DatabaseName
+  private val CollectionName = "notifications"
+
+  private implicit val mongoDatabase: MongoDatabase = {
+    val uri = new MongoClientURI(MongoURI.replaceAllLiterally("sslEnabled", "ssl"))
+    val client = new MongoClient(uri)
+
+    client.getDatabase(DatabaseName)
+  }
+
+  private val changeLog = new MakeParsedDetailsOptional()
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    mongoDatabase.getCollection(CollectionName).drop()
+  }
+
+  override def afterEach(): Unit = {
+    mongoDatabase.getCollection(CollectionName).drop()
+    super.afterEach()
+  }
+
+  private def getDeclarationsCollection(db: MongoDatabase): MongoCollection[Document] = mongoDatabase.getCollection(CollectionName)
+
+  "MakeParsedDetailsOptional migration definition" should {
+
+    "correctly migrate records from previous format" in {
+      runTest(testDataBeforeChangeSet_1, testDataAfterChangeSet_1)(changeLog.migrationFunction)
+    }
+
+    "not change records already migrated" in {
+      runTest(testDataAfterChangeSet_1, testDataAfterChangeSet_1)(changeLog.migrationFunction)
+    }
+
+    "not change records that could not be parsed" in {
```

Comment:
For some reason this test name mislead me. At first glance I thought it is stating that we are not going to migrate what we cannot parse. 
It is not super-crucial, but what do you think about "not change records that were not yet parsed" ?

Reply:
Ok, Renamed